packages/flutter/test/cupertino/dialog_test.dart

```diff
@@ -1002,27 +1002,27 @@ void main() {
 
     // Exit animation, look at reverse FadeTransition.
     await tester.pump(const Duration(milliseconds: 25));
-    transition = tester.widgetList(find.byType(FadeTransition)).elementAt(1);
+    transition = tester.widgetList(find.byType(FadeTransition)).elementAt(1) as FadeTransition;
```

Comment:
Here and elsewhere for `tester.widgetList`: do we still need the cast if we specify the type parameter? `tester.widgetList<FadeTransition>(find.byType(FadeTransition))`?  I'm not 100% sure this actually avoids a cast, but I think we prefer using type parameters when we can anyway.

Reply:
Disabling `implicit-casts` makes me wondering how to improve type safety. I think some APIs could be improved by making them generics but I wanted to use ` as Xxx` before looking at the changes that could perhaps be done (and I suspect it could be breaking changes)